src/main/java/org/gridsuite/study/server/NetworkModificationTreeService.java

```diff
@@ -448,4 +474,19 @@ public void doUpdateBuildStatus(UUID nodeUuid, BuildStatus buildStatus) {
     public BuildStatus getBuildStatus(UUID nodeUuid) {
         return nodesRepository.findById(nodeUuid).map(n -> repositories.get(n.getType()).getBuildStatus(nodeUuid)).orElse(BuildStatus.NOT_BUILT);
     }
+
+    @Transactional(readOnly = true)
+    public Optional<UUID> doGetParentNode(UUID nodeUuid, NodeType nodeType) {
+        NodeEntity nodeEntity = nodesRepository.findById(nodeUuid).orElseThrow(() -> new StudyException(ELEMENT_NOT_FOUND));
+        if (nodeEntity.getType() == NodeType.ROOT || nodeEntity.getParentNode().getType() == nodeType) {
+            return Optional.of(nodeEntity.getParentNode().getIdNode());
+        } else {
+            return doGetParentNode(nodeEntity.getParentNode().getIdNode(), nodeType);
```

Comment:
that means that we make one request per node until we get the type requested...

Reply:
Maybe, an optimization could be done later ...